jdk/make/closed/autoconf/custom-hook.m4

```diff
@@ -702,6 +702,21 @@ AC_DEFUN([CONFIGURE_OPENSSL],
 
     AC_MSG_CHECKING([if we should bundle openssl])
     AC_MSG_RESULT([$BUNDLE_OPENSSL])
+
+    if test "x$OPENJ9_ENABLE_JITSERVER" = xtrue ; then
+      if test "x$OPENJDK_TARGET_OS" = xlinux ; then
+        if test "x$OPENSSL_CFLAGS" != x; then
+          CHECK_OPENSSL_HEADER=`$ECHO $OPENSSL_CFLAGS | $SED -e 's/\-I//'`
+          CHECK_OPENSSL_HEADER="${CHECK_OPENSSL_HEADER}/openssl/ssl.h"
+          AC_MSG_CHECKING([if the required OPENSSL API exists for JITServer in $CHECK_OPENSSL_HEADER])
+          if $GREP SSL_CTX_set_ecdh_auto "$CHECK_OPENSSL_HEADER" 2>&1 > /dev/null ; then
+            AC_MSG_RESULT([yes])
+          else
+            AC_MSG_ERROR([no, SSL_CTX_set_ecdh_auto is required by JITServer])
+          fi
+        fi
```

Comment:
* The check should occur even if `OPENSSL_CFLAGS` is empty (meaning no special compile options are necessary) and it's unreasonable to assume that `OPENSSL_CFLAGS` consists only of a single `-I` option: You should just use `$OPENSSL_DIR/include/openssl/ssl.h`.

* Both branches of the `if` should have `AC_MSG_RESULT`.

```
  AC_MSG_CHECKING([existence of OPENSSL API required by JITServer in openssl/ssl.h])
  if $GREP -qw SSL_CTX_set_ecdh_auto "$OPENSSL_DIR/include/openssl/ssl.h" 2> /dev/null ; then
    AC_MSG_RESULT([yes])
  else
    AC_MSG_RESULT([no])
    AC_MSG_ERROR([SSL_CTX_set_ecdh_auto is required by JITServer])
  fi
```
* Also notice the use of grep options `-q` and `-w`, and the change to stderr redirection.

Reply:
* `OPENSSL_DIR` is not set when `--with-openssl=system` is specified but `OPENSSL_CFLAGS` will get set.
* Will add `AC_MSG_RESULT([no])`. 
* Will change it to `if $GREP -q -w SSL_CTX_set_ecdh_auto "$CHECK_OPENSSL_HEADER" ;`

Reply:
In the `--with-openssl=system` case, I would expect some variable to be set by `PKG_CHECK_MODULES(OPENSSL, ...)` that can be use to set `OPENSSL_DIR ` (or that variable should be used throughout instead of `OPENSSL_DIR`).

Reply:
[1] indicates `OPENSSL_CFLAGS` and `OPENSSL_LIB` will be set up by `PKG_CHECK_MODULES`. My test shows `cmake version 3.5.1` will set them up but `cmake version 3.10.2` will not with our current `jdk/make/closed/autoconf/custom-hook.m4`. Using `OPENSSL_CFLAGS` here just gives more cases to check this one corner case of `SSL_CTX_set_ecdh_auto`. 

[1]
https://github.com/ibmruntimes/openj9-openjdk-jdk8/blob/8514ba3391a835583387d5065fccd009c13379c6/jdk/make/closed/autoconf/custom-hook.m4#L629

Reply:
Both versions of cmake set `OPENSSL_INCLUDE_DIR` which seems like the right place to look for header files and the right name to use when `--with-openssl` is not `system`.

Reply:
Sorry, I mixed up the `autoconf PKG_CHECK_MODULES` with the` cmake pkg_check_modules`. We are using `autoconf PKG_CHECK_MODULES` in `custom-hook.m4`. 

`PKG_CHECK_MODULES` sets only two variables `<>_CFLAGS` and `<>_LIBS` [1]. 

There are also two scenarios to consider when setting `--with-openssl=system`. 

1. If OpenSSL is installed at the default system location such as `/usr`, `pkg-config` could strip out the flags. `<>_CFLAGS` and `<>_LIBS` will not be set by `PKG_CHECK_MODULES` [2]. 

Running bash configure as below will NOT set  `<>_CFLAGS` and `<>_LIBS`:
```
bash configure --with-freemarker-jar=/root/freemarker.jar --with-boot-jdk=/root/bootjdk8 --enable-jitserver --with-openssl=system` 
```

Running bash configure with `PKG_CONFIG_ALLOW_SYSTEM_CFLAGS` and  `PKG_CONFIG_ALLOW_SYSTEM_LIBS ` will have both `<>_CFLAGS` and `<>_LIBS` set:
```
bash configure --with-freemarker-jar=/root/freemarker.jar --with-boot-jdk=/root/bootjdk8 --enable-jitserver --with-openssl=system PKG_CONFIG_ALLOW_SYSTEM_CFLAGS=1 PKG_CONFIG_ALLOW_SYSTEM_LIBS=1``` 
...
OPENSSL_CFLAGS=-I/usr/include
OPENSSL_LIBS=-L/usr/lib/x86_64-linux-gnu -lssl -lcrypto
``` 

2. If OpenSSL is installed at non default locations such as `/usr/local`, `PKG_CHECK_MODULES` will set `<>_CFLAGS` and `<>_LIBS`
```
bash configure --with-freemarker-jar=/root/freemarker.jar --with-boot-jdk=/root/bootjdk8 --enable-jitserver --with-openssl=system
...
OPENSSL_CFLAGS=-I/usr/local/include
OPENSSL_LIBS=-L/usr/local/lib -lssl -lcrypto
```
---------------------------------------------------------------------------
[1] https://github.com/ibmruntimes/openj9-openjdk-jdk8/blob/8514ba3391a835583387d5065fccd009c13379c6/common/autoconf/build-aux/pkg.m4#L128-L131
[2] https://linux.die.net/man/1/pkg-config